cpp/src/arrow/ipc/message.cc

```diff
@@ -43,7 +43,12 @@ class Message::MessageImpl {
       : metadata_(metadata), message_(nullptr), body_(body) {}
 
   Status Open() {
-    message_ = flatbuf::GetMessage(metadata_->data());
+    auto data = metadata_->data();
+    flatbuffers::Verifier verifier(data, metadata_->size(), /*max_depth=*/128);
+    if (!flatbuf::VerifyMessageBuffer(verifier)) {
+      return Status::IOError("Invalid flatbuffers message.");
+    }
+    message_ = flatbuf::GetMessage(data);
```

Comment:
Can you factor this into a helper function to avoid code duplication?

Reply:
helper function doesn't work due to the early `IOError` return

Reply:
And honestly I was not able to find a nice macro that does this. The issue is basically that you would need to cover 2 things: the early exit and the assignment to `message_` (or another variable)